pkg/cache/repository.go

```diff
@@ -472,25 +472,8 @@ func (r *cachedRepository) refreshAllCachedPackages(ctx context.Context) (map[re
 		}
 	}
 
-	// We go through all the PackageRevisions and make sure they have
-	// a corresponding PackageRev CR.
-	for pkgRevName, pkgRev := range newPackageRevisionNames {
-		if _, found := existingPkgRevCRsMap[pkgRevName]; !found {
-			pkgRevMeta := meta.PackageRevisionMeta{
-				Name:      pkgRevName,
-				Namespace: r.repoSpec.Namespace,
-			}
-			if _, err := r.metadataStore.Create(ctx, pkgRevMeta, r.repoSpec.Name, pkgRev.UID()); err != nil {
-				// TODO: We should try to find a way to make these errors available through
-				// either the repository CR or the PackageRevision CR. This will be
-				// retried on the next sync.
-				klog.Warningf("unable to create PackageRev CR for %s/%s: %v",
-					r.repoSpec.Namespace, pkgRevName, err)
-			}
-		}
-	}
-
 	// Send notification for packages that changed.
+	// Send notification first before creation of PkgRev to avoid race condition after an approve because of ownerReference.
```

Comment:
```suggestion
	// Send notification for packages that changed before creation of PkgRev to avoid race condition after an approve because of ownerReferences.
```
```suggestion
	// Send notification for packages that changed.
	// Send notification first before creation of PkgRev to avoid race condition after an approve because of ownerReference.
```
```suggestion
	// Send notification for packages that changed before the creation of PkgRev to avoid race conditions because of ownerReferences.
```